src/metrics/list_repos.rs

```diff
@@ -62,66 +64,56 @@ impl Producer for ListReposForOrg {
+#[derive(Deserialize, Debug)]
+struct Data {
+    search: IssueCount,
+}
+
+#[derive(Deserialize, Debug)]
+struct IssueCount {
+    #[serde(rename = "issueCount")]
+    issue_count: u32,
+}
+
+/// count the number of pull requests created in the given time period for the given repository within the given GitHub organization
 ///
 /// # Arguments
-///
-/// - `octo` — The instance of `octocrab::Octocrab` that should be used to make queries to GitHub API
 /// - `org_name` — The name of the github organization that owns the specified repository
 /// - `repo_name` — The name of the repository to count pull requests for. **Note:** repository should exist within the `org_name` Github Organization
-///
-/// # Example
-///
-/// ```
-/// use github-metrics;
-/// use octocrab;
-/// use std::string::String;
-///
-/// let octocrab_instance = octocrab::Octocrab::builder().personal_token("SOME_GITHUB_TOKEN").build()?;
-///
-/// const num_pull_requests = github-metrics::count_pull_requests(octocrab_instance, "rust-lang", "rust");
-///
-/// println!("The 'rust-lang/rust' repo has had {} Pull Requests created in the last 30 days!", num_pull_requests);
-/// ```
+/// - `time_period` — The relevant time period to search within
 #[throws]
-async fn count_pull_requests(org_name: &String, repo_name: &str) -> usize {
-    let octo = octocrab::instance();
-    let mut page = octo
-        .pulls(org_name, repo_name)
-        .list()
-        // take all PRs -- open or closed
-        .state(octocrab::params::State::All)
-        // sort by date created
-        .sort(octocrab::params::pulls::Sort::Created)
-        // start with most recent
-        .direction(octocrab::params::Direction::Descending)
-        .per_page(255)
-        .send()
-        .await?;
-
-    let thirty_days_ago = Utc::now() - Duration::days(30);
-    let mut pr_count: usize = 0;
+async fn count_pull_requests_graphql(
+    org_name: &str,
+    repo_name: &str,
+    time_period: Duration,
+) -> usize {
+    // the following madness simply removes the "UTC" at the end of the
+    // date string to match GitHub's PR query format for `created` field
+    // i.e., "2021-05-18UTC" turns into "2021-05-18"
+    let date_string = format!("{}", (Utc::now() - time_period).date());
+    let mut chars = date_string.chars();
+    for _ in 0..3_u8 {
+        chars.next_back();
+    }
+    let thirty_days_ago_str = chars.as_str();
 
-    loop {
-        let in_last_thirty_days = page
-            .items
-            .iter()
-            // take while PRs have been created within the past thirty days
-            .take_while(|pr| pr.created_at > thirty_days_ago)
-            .count();
+    let query_string = format!(
+        r#"query {{
+            search(
+                query:"repo:{org_name}/{repo_name} is:pr created:>{thirty_days_ago}",
+                type:ISSUE,
+                last:100,
```

Comment:
I'm confused by this code -- what does `last:100` do? I would expect that to only count up to 100 issues or something

Reply:
This is not an issue for this query in `count_pull_requests_graphql` because we don't actually care about the pages of data (for now). (one of the reasons it's so fast, probably)

Take this query, for example: 
```graphql
{
  search(
    query: "repo:rust-lang/rust is:pr created:>2021-05-19", type: ISSUE, last: 100
  ) {
    issueCount
    pageInfo {
      hasNextPage
      hasPreviousPage
      startCursor
      endCursor
    }
  }
}
```

it returns the following data: 
```json
{
  "data": {
    "search": {
      "issueCount": 588,
      "pageInfo": {
        "hasNextPage": true,
        "hasPreviousPage": false,
        "startCursor": "Y3Vyc29yOjE=",
        "endCursor": "Y3Vyc29yOjEwMA=="
      }
    }
  }
}
```

The reason this is such an improvement is that `issueCount` is a native property on whatever GQL type the `search` query responds with. And since that's (for now) the only item we care about—we could go to each next page, but that `issueCount` would stay the same at 588 for each page. Performance is so much better since we're not walking in to the `edges { node {}}` for every single PR.

Reply:
That said, I'll go ahead and change it to `last: 1` in case that helps performance in any marginal way. 

And I'll leave a comment to make the purpose behind this more clear.